_posts/2018-06-13-Serverless-Basic-Authentication-Custom-Authorizer.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: post
+title: Serverless Basic Authentication using a Custom Authorizer
+banner: /assets/posts/2018-06-13-Serverless-Basic-Authentication-Custom-Authorizer/banner.png
+github: svdgraaf/serverless-basic-authentication
+author: svdgraaf
+---
+
+In a recent project, we needed our api's to be able to work with external systems. These systems only supported http basic authentication (eg: username/password) for integrating with external systems.
+
+Of course, Basic HTTP Authentication is the easiest and most straight forward way to integrate things and comes build in, in almost any http client.
+
+If you want to read more about basic authentication, I suggest you take a look at the [wikipedia page](https://en.wikipedia.org/wiki/Basic_access_authentication) or [RFC7617](https://tools.ietf.org/html/rfc7617).
+
+Api Gateway
+-----------
+HTTP Basic Authentication is a very simple standard, which is why almost everyone supports it. Except for Api Gateway. The easiest way to setup authentication on Api Gateway is either to use Api Keys or IAM authentication. Using IAM authentication makes you have to sign all requests. Using the Api Keys, you need to add a header (`x-api-key` to each request, with the api key in it).
```

Comment:
"HTTP Basic Authentication is a very simple standard, which is why almost everyone supports it. Except for Api Gateway."

->

"HTTP Basic Authentication is a very simple standard, which is why almost everyone supports it. But not Api Gateway."